pints/_mcmc/_mala.py

```diff
@@ -0,0 +1,319 @@
+    discretisation, a Metropolis-Hastings acceptance probability is
+    calculated after a step,
+
+    .. math::
+        \\alpha = \\frac{\\pi(\\theta^*)q(\\theta_t|\\theta^*)}{\\pi(\\theta^*)
+            q(\\theta^*|\\theta_t)}
+
+    where :math:`q(\\theta_2|\\theta_1) =
+        \\mathcal{N}(\\theta_2|\\mu(\\theta_1), \\epsilon I)` and
+    :math:`\\theta^*` is accepted with probability
+    :math:`\\text{min}(1, \\alpha)`.
+
+    Here we consider a slight variant of the above method discussed in [1],
+    which is to use a preconditioning matrix :math:`M` to allow differing
+    degrees of freedom in each dimension.
+
+    .. math::
+        \\theta^* = \\theta_t + \\epsilon'^2 1/2 \\nabla \\;
+            \\text{log} \\pi(\\theta_t) + \\epsilon' z
+
+    leading to :math:`q(\\theta_2|\\theta_1) =
+        \\mathcal{N}(\\theta_2|\\mu(\\theta_1), \\epsilon')`.
+
+    where :math:`\\epsilon' = \\epsilon sqrt{M}` is given by the initial
+    value of `sigma0`.
+
+    *Extends:* :class:`SingleChainMCMC`
+
+    [1] Girolami, M. and Calderhead, B., 2011. Riemann manifold langevin and
+    hamiltonian monte carlo methods. Journal of the Royal Statistical Society:
+    Series B (Statistical Methodology), 73(2), pp.123-214.
+    """
+
+    def __init__(self, x0, sigma0=None):
+        super(MALAMCMC, self).__init__(x0, sigma0)
+
+        # Set initial state
+        self._running = False
+        self._ready_for_tell = False
+
+        # Current point and proposed point
+        self._current = None
+        self._current_log_pdf = None
+        self._proposed = None
+
+        # hyper parameters
+        self._epsilon = None
+        self._step_size = 0.1
+        self.set_scale_vector(np.diag(self._sigma0))
+
+    def _initialise(self):
+        """
+        Initialises the routine before the first iteration.
+        """
+        if self._running:
+            raise RuntimeError('Already initialised.')
+
+        # Propose x0 as first point
+        self._current = None
+        self._current_log_pdf = None
+        self._proposed = self._x0
+
+        # Acceptance rate monitoring
+        self._iterations = 0
+        self._acceptance = 0
+
+        # Update sampler state
+        self._running = True
+
+        # Initialise step size
+        self._forward_mu = None
+        self._backward_mu = None
+        self._current_gradient = None
+        self._proposed_gradient = None
+        self._forward_q = None
+        self._backward_q = None
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def set_step_size(self, step_size=0.1):
+        """
+        Sets step size used to propose new points. Must exceed 0
+        """
+        step_size = float(step_size)
+        if step_size <= 0:
+            raise ValueError('Step size must exceed 0')
+        self._step_size = step_size
+        self.set_epsilon()
```

Comment:
This will unset a manually specified epsilon!

It might be better to leave epsilon unset (e.g. `None` unless the user sets it explicitly), and then only set it automatically in initialise (if it's still None at that time)

Reply:
@MichaelClerx Thanks for these points. It was far too complicated before. Now I have removed the set_scale and set_step_size functions as we only need epsilon.